scripts/static-vdis

```diff
@@ -230,13 +244,40 @@ def attach(vdi_uuid):
                 return d + "/disk"
     if not found:
         raise Exception("Disk configuration not found")
+
+def detach(vdi_uuid):
+    found = False
+    for existing in list():
+        if existing['vdi-uuid'] == vdi_uuid:
+            if not (existing.has_key('disk')):
+                return
+            found = True
+            d = main_dir + "/" + existing['id']
+            if not (os.path.exists(d + "/sr-uri")):
+                # SMAPIv1
+                config = read_whole_file(d + "/config")
+                driver = read_whole_file(d + "/driver")
+                call_backend_detach(driver, config)
+            else:
+                volume_plugin = read_whole_file(d + "/volume-plugin")
```

Comment:
For consistency, I think these should be either called `volume_{plugin,key,uri}` or `plugin`, `key`, `uri` but not using different prefixes.